src/finch/finch_logic/_utils.py

```diff
@@ -0,0 +1,27 @@
+from collections.abc import Iterable
+
+from .nodes import Field
+
+
+class NonConcordantLists(Exception):
+    pass
+
+
+def merge_concordant(args: Iterable[Iterable[Field]]) -> list[Field]:
```

Comment:
Here we merge all `args` lists so that the merged lists retains the order of all constituent lists. Raises an error if inputs aren't concordant.